### PR TITLE
feat: homepage and testing of homepage

### PR DESCRIPTION
**Pull request by Ajanyaat**
I merge the code and resolve the conflicts.

### CODE REVIEW

File: diagramming-tool/package-lock.json
Comment (AnushiyaM11) [inline diagramming-tool/package-lock.json:1]: Please ignore the package-lock.json file while committing the changes to the branch

File: diagramming-tool/src/App.test.js
Comment (AnushiyaM11) [inline diagramming-tool/src/App.test.js:1]: Remove all the commented lines of code

File: diagramming-tool/src/App.js
Comment (AnushiyaM11) [inline diagramming-tool/src/App.js:1]: The React component file should be .jsx

File: diagramming-tool/src/Component/HomePage/.babelrc
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/HomePage/.babelrc:1]: .babelrc file should be included in the root folder

File: diagramming-tool/package-lock.json
Comment (Ajanyaat) [reply diagramming-tool/package-lock.json:1]: Done Changes

File: diagramming-tool/public/OIP (1).jpeg
[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/public/OIP (1).jpeg:1]: Done changes

File: diagramming-tool/src/App.test.js
Comment (Ajanyaat) [reply diagramming-tool/src/App.test.js:1]: Done Changes

File: diagramming-tool/src/App.js
Comment (Ajanyaat) [reply diagramming-tool/src/App.js:1]: Done Changes

File: diagramming-tool/src/Component/HomePage/.babelrc
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/.babelrc:1]: Done Project

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+
+
+
+import React, { useState } from 'react';
+import './HomePage.css'; 
+import diagImage from './diag.png';
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/HomePage/HomePage.js:6]: Provide a meaningful name

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import exampleImage from './example.jpg';
+import peopleImage from './people.png';
+import featuresImage from './features.png';
+import keImage from './ke.png'; 
+import dig from './dig.svg';
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/HomePage/HomePage.js:12]: Provide a meaningful name

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        </button>
+        {showOptions && (
+          <div className="options-box" data-testid="options-box">
+            <p className="option-description"></p>
+          </div>
+        )}
+      </section>
+      <img src={exampleImage} alt="Example" className="example-image" /> {/* Example image */}
+      <img src={peopleImage} alt="People" className="bottom-left-image" />
+      <img src={featuresImage} alt="Features" className="bottom-leeft-image" />
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/HomePage/HomePage.js:42]: Check the spelling for ClassName

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  return (
+    <div className="home-page-container" data-testid="home-page-container"> {/* Add data-testid here */}
+      <img src={diagImage} alt="Diagram" className="diag-image" />
+      <header className="header">
+        <h1 className="title">Welcome to Our Diagramming Tool</h1>
+        <p className="subtitle">Visualize your ideas and concepts with ease</p>
+        {/* Move this inside */}
+        <nav className="nav"></nav>
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/HomePage/HomePage.js:28]: Usage of <nav> ? It is showing empty here.

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        {showOptions && (
+          <div className="options-box" data-testid="options-box">
+            <p className="option-description"></p>
+          </div>
+        )}
+      </section>
+      <img src={exampleImage} alt="Example" className="example-image" /> {/* Example image */}
+      <img src={peopleImage} alt="People" className="bottom-left-image" />
+      <img src={featuresImage} alt="Features" className="bottom-leeft-image" />
+      <img src={keImage} alt="KE" className="bottom-leeeft-image" />
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/HomePage/HomePage.js:43]: ClassName

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        <h2 className="bottom-left-title3">Privacy First</h2>
+        <p className="bottom-left-subtitle3">Store your data wherever you want to. We cannot access your data</p>
+        {/* New sentences */}
+        <div className="bottom-left-additional">
+          <p>Experience the seamless integration of collaboration and visualization, as you co-create with colleagues in real-time, regardless of distance or time zone.</p>
+          <p>Embark on a journey of discovery and ideation as you explore the endless possibilities our tool offers for expressing and refining your thoughts.</p>
+          <p>From brainstorming to presentation, our comprehensive suite of tools empowers you at every stage of your creative process, ensuring your ideas are brought to life with clarity and precision.</p>
+        </div>
+      </div>
+      {
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/HomePage/HomePage.js:63]: The footer JSX is already a JSX expression, so there's no need to wrap it with curly braces

File: diagramming-tool/src/Component/HomePage/HomePage.js (hunk 1 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: done changes

File: diagramming-tool/src/Component/HomePage/HomePage.js (hunk 2 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: done changes

File: diagramming-tool/src/Component/Home/Home.js [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+      height: 100,
+      radius: 50,
+      size: 80,
+    };
+    setShapes([...shapes, newShape]);
+    setSelectedShape(newShape.id);
+  };
+// 
+  const handleUndo = () => {
+  };
```
Comment (Sharz-dev) [inline diagramming-tool/src/Component/Home/Home.js:163]: Complete the logic for undo  save,delete,Redo

File: diagramming-tool/src/Component/HomePage/HomePage.js (hunk 3 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: done changes

File: diagramming-tool/src/Component/HomePage/HomePage.js (hunk 5 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: done changes

File: diagramming-tool/src/Component/HomePage/HomePage.js (hunk 4 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: done changes

File: diagramming-tool/src/Component/HomePage/HomePage.js (hunk 6 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: done changes

File: diagramming-tool/public/index.html
Comment (AnushiyaM11) [inline diagramming-tool/public/index.html:1]: Are you sure you wouldn't need these script and link

File: diagramming-tool/src/ApiService/ApiService.jsx
Comment (AnushiyaM11) [inline diagramming-tool/src/ApiService/ApiService.jsx:1]: Is this file ok to be deleted as it has API code for registration

File: diagramming-tool/src/App.css
Comment (AnushiyaM11) [inline diagramming-tool/src/App.css:1]: Could you please confirm that this file is no longer needed

File: diagramming-tool/src/App.js
Comment (AnushiyaM11) [inline diagramming-tool/src/App.js:1]: Router changes should be included in this file

File: diagramming-tool/src/Component/ConfirmMsg/MsgBoxComponent.jsx
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/ConfirmMsg/MsgBoxComponent.jsx:1]: Please confirm if this files can be deleted

File: diagramming-tool/src/Component/ConfirmMsg/MsgComponent.css
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/ConfirmMsg/MsgComponent.css:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Component/HomePage/HomePage.js
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/HomePage/HomePage.js:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Component/Login/LoginPage.css
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Login/LoginPage.css:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Component/Login/LoginPage.jsx
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Login/LoginPage.jsx:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Component/Register/Registration.css
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Register/Registration.css:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Component/Register/Registration.jsx
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Register/Registration.jsx:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Component/Register/Registration.test.js
Comment (AnushiyaM11) [inline diagramming-tool/src/Component/Register/Registration.test.js:1]: Please confirm if this file can be deleted

File: diagramming-tool/src/Images/background.jpeg.jpg
Comment (AnushiyaM11) [inline diagramming-tool/src/Images/background.jpeg.jpg:1]: Please move the images to assets folder and not images

**Review by AnushiyaM11 — COMMENTED** [review]
Please check the comments and do the needful changes

File: diagramming-tool/src/Images/background.jpeg.jpg
Comment (Ajanyaat) [reply diagramming-tool/src/Images/background.jpeg.jpg:1]: changes done

File: diagramming-tool/src/Component/HomePage/HomePage.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+const HomePage = () => {
+  const [showOptions, setShowOptions] = useState(false);
+
+  const toggleOptions = () => {
+    setShowOptions(!showOptions);
+  };
+
+  return (
+    <div className="home-page-container" data-testid="home-page-container"> {/* Add data-testid here */}
```
[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js]: changes done

File: diagramming-tool/src/Component/HomePage/background.jpeg.jpg
[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/background.jpeg.jpg:1]: changes done

File: diagramming-tool/src/Component/HomePage/HomePage.js
[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/HomePage.js:1]: done changes

File: diagramming-tool/src/Component/HomePage/jest.config.js
[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/src/Component/HomePage/jest.config.js:1]: done changes

File: diagramming-tool/public/index.html
Comment (Ajanyaat) [reply diagramming-tool/public/index.html:1]: No need .

File: diagramming-tool/src/Component/Login/LoginPage.jsx
Comment (Ajanyaat) [reply diagramming-tool/src/Component/Login/LoginPage.jsx:1]: Yes I can delete

File: diagramming-tool/src/Component/Register/Registration.css
Comment (Ajanyaat) [reply diagramming-tool/src/Component/Register/Registration.css:1]: Yes I can delete

File: diagramming-tool/src/Component/Register/Registration.jsx
Comment (Ajanyaat) [reply diagramming-tool/src/Component/Register/Registration.jsx:1]: Yes I can delete

File: diagramming-tool/src/Component/Register/Registration.test.js
Comment (Ajanyaat) [reply diagramming-tool/src/Component/Register/Registration.test.js:1]: Yes I can delete

File: diagramming-tool/public/index.html [hunk 9]
```diff
@@ -14,10 +14,6 @@
       manifest.json provides metadata used when your web app is installed on a
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
-    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.4.3/dist/css/bootstrap.min.css" rel="stylesheet">
-    <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js"></script>
```
[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/public/index.html:18]: I didn't change anything in this file .

[the remark this answers is not in the document]
Comment (Ajanyaat) [reply diagramming-tool/public/index.html:18]: We didn't use jquery and bootstrap.

File: diagramming-tool/src/Component/Home/Home.js (hunk 7 above)
Comment (Ajanyaat) [reply diagramming-tool/src/Component/Home/Home.js]: Done changes

**Review by Sharz-dev — APPROVED** [review]

**Pull request merged**